internal/resource/config/oauth/oauth_auth_server_settings_resource.go

```diff
@@ -0,0 +1,1167 @@
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{"INTERNAL", "ADAPTER"}...),
+				},
+			},
+			"user_authorization_consent_adapter": schema.StringAttribute{
+				Description: "Adapter ID of the external consent adapter to be used for the consent page user interface.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"approved_scopes_attribute": schema.StringAttribute{
+				Description: "Attribute from the external consent adapter's contract, intended for storing approved scopes returned by the external consent page.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"approved_authorization_detail_attribute": schema.StringAttribute{
+				Description: "Attribute from the external consent adapter's contract, intended for storing approved authorization details returned by the external consent page.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"par_reference_timeout": schema.Int64Attribute{
+				Description: "The timeout, in seconds, of the pushed authorization request reference. The default value is 60.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
+			},
+			"par_reference_length": schema.Int64Attribute{
+				Description: "The entropy of pushed authorization request references, in bytes. The default value is 24.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
+			},
+			"par_status": schema.StringAttribute{
+				Description: "The status of pushed authorization request support. The default value is ENABLED.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+				Validators: []validator.String{
+					stringvalidator.OneOf([]string{"DISABLED", "ENABLED", "REQUIRED"}...),
+				},
+			},
+			"client_secret_retention_period": schema.Int64Attribute{
+				Description: "The length of time in minutes that client secrets will be retained as secondary secrets after secret change. The default value is 0, which will disable secondary client secret retention.",
+				Computed:    true,
+				Optional:    true,
+				PlanModifiers: []planmodifier.Int64{
+					int64planmodifier.UseStateForUnknown(),
+				},
+			},
+			"jwt_secured_authorization_response_mode_lifetime": schema.Int64Attribute{
+				Description: "The lifetime, in seconds, of the JWT Secured authorization response. The default value is 600.",
```

Comment:
It looks like we know the defaults for a lot of these attributes, so it would be nice to set them in the schema here to have more consistent provider runs across versions.